solr/solrj/src/java/org/apache/solr/common/cloud/ZkStateReader.java

```diff
@@ -2269,7 +2269,12 @@ public boolean update() throws KeeperException, InterruptedException {
       log.debug("Checking ZK for most up to date Aliases {}", ALIASES);
       // Call sync() first to ensure the subsequent read (getData) is up to date.
       zkClient.getSolrZooKeeper().sync(ALIASES, null, null);
-      Stat stat = new Stat();
+      Stat stat = zkClient.exists(ALIASES, null, true);
+      if (stat.getVersion() <= aliases.getZNodeVersion()) {
```

Comment:
Interesting.  So you've found that it's faster to request only the "Stat" without the data so long as this is the typical path?